### ISSUE TITLE
Problem with  xe_costs_en_fr_en loss when reloading checkpoint

### ISSUE THREAD

**Issue by pbordes** [facebookresearch/UnsupervisedMT#47]
Hello, 
I have a problem when I stop  then restart training ... When I stop training, I have a xe_costs_en_fr_en loss value of about 0.8, but when I restart training, the value is 6; even if the checkpoint has been successfully reloaded, this back-translation loss seems to begin from scratch again (the other ones are ok). Would you know why ?( I am asking this because I'd like to add some complementary losses at certain times of training )
Thank you very much!

**Comment by glample** [comment]
Hi,

How are you doing the reload? Are you reloading a pretrained model? Or a `checkpoint.pth` file? In the first case, what you observe is possible (even though the difference seems surprisingly large), since the optimizer weights will also be reset, but in the case of a checkpoint, the loss should be the same before and after the training interruption, since the optimizer is also reloaded.

Maybe could you try to print the weights before and after reloading (both for the model and the optimizer) to see if they are identical?

**Comment by pbordes** [comment]
Hello, 
Thank you for your quick answer ! I trained entirely a model (until the stopping criterion) and now I am trying to reload it: do you mean that when a model is trained until the end then the optimizer weights are not saved ? However, the penultimate function called in the main.py is end_epoch() which calls save_checkpoint(), and this function saves optimizer weights ...

**Comment by glample** [comment]
What is the name of the reloaded file? If it's something like `best-bleu_en_fr.pth` it will only contain the model. The `checkpoint.pth` file is to reload experiments that have crashed or have been interrupted for various reasons, and for which you want to resume the training.

**Comment by pbordes** [comment]
Ok I got it! thank you very much
